engine/src/vs_logging.cpp

```diff
@@ -46,6 +45,103 @@
+#include "opentelemetry/sdk/trace/simple_processor_factory.h"
+#include "opentelemetry/sdk/trace/tracer_provider.h"
+#include "opentelemetry/sdk/trace/tracer_provider_factory.h"
+#include "opentelemetry/trace/tracer_provider.h"
+
+namespace nostd = opentelemetry::nostd;
+
+namespace
+{
+opentelemetry::exporter::otlp::OtlpFileExporterOptions opts;
+opentelemetry::exporter::otlp::OtlpFileLogRecordExporterOptions log_opts;
+
+std::shared_ptr<opentelemetry::sdk::trace::TracerProvider> tracer_provider;
+std::shared_ptr<opentelemetry::sdk::logs::LoggerProvider> logger_provider;
+
+void InitTracer()
+{
+    // Create OTLP exporter instance
+    auto exporter   = opentelemetry::exporter::otlp::OtlpFileExporterFactory::Create(opts);
+    auto processor  = opentelemetry::sdk::trace::SimpleSpanProcessorFactory::Create(std::move(exporter));
+    tracer_provider = opentelemetry::sdk::trace::TracerProviderFactory::Create(std::move(processor));
+
+    // Set the global trace provider
+    std::shared_ptr<opentelemetry::trace::TracerProvider> api_provider = tracer_provider;
+    opentelemetry::trace::Provider::SetTracerProvider(api_provider);
+}
+
+void CleanupTracer()
+{
+  // We call ForceFlush to prevent to cancel running exportings, It's optional.
+  if (tracer_provider)
+  {
+    tracer_provider->ForceFlush();
+  }
+
+  tracer_provider.reset();
+  const nostd::shared_ptr<opentelemetry::trace::TracerProvider> none;
+  opentelemetry::trace::Provider::SetTracerProvider(none);
+}
+
+void InitLogger()
+{
+  // Create OTLP exporter instance
+  auto exporter   = opentelemetry::exporter::otlp::OtlpFileLogRecordExporterFactory::Create(log_opts);
+  auto processor  = opentelemetry::sdk::logs::SimpleLogRecordProcessorFactory::Create(std::move(exporter));
+  logger_provider = opentelemetry::sdk::logs::LoggerProviderFactory::Create(std::move(processor));
+
+  std::shared_ptr<opentelemetry::logs::LoggerProvider> api_provider = logger_provider;
+  opentelemetry::logs::Provider::SetLoggerProvider(api_provider);
+}
+
+void CleanupLogger()
+{
+  // We call ForceFlush to prevent to cancel running exportings, It's optional.
+  if (logger_provider)
+  {
+    logger_provider->ForceFlush();
+  }
+
+  logger_provider.reset();
+  nostd::shared_ptr<opentelemetry::logs::LoggerProvider> none;
+  opentelemetry::logs::Provider::SetLoggerProvider(none);
+}
+
+nostd::shared_ptr<opentelemetry::trace::Tracer> get_tracer()
+{
+    const nostd::shared_ptr<opentelemetry::trace::TracerProvider> provider = opentelemetry::trace::Provider::GetTracerProvider();
+    return provider->GetTracer(__FILE__);
+}
+
+nostd::shared_ptr<opentelemetry::logs::Logger> get_logger()
+{
+    const nostd::shared_ptr<opentelemetry::logs::LoggerProvider> provider = opentelemetry::logs::Provider::GetLoggerProvider();
+    return provider->GetLogger(__FILE__);
+}
+
+}  // namespace
+
+#endif
+
 namespace VegaStrikeLogging {
 
 void VegaStrikeLogger::InitLoggingPart2(const uint8_t debug_level,
```

Comment:
Not a big fan of init2. Is there anything we can do here?
Also, I don't see where this is called from. This may be a limitation of the GitHub web interface.

Yes, I know I've done it as well.
I can only plead that I was a very poor C++ programmer back then.